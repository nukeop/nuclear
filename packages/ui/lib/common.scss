$bgdarker: #1F212D;
$bgdark: #2C2E3A;
$bgdefault: #424557;
$bglight: #555977;
$bglighter: #6A6F95;

$background: $bgdark;
$background2: $bgdefault;
$background3: $bgdarker;
$lightbg: $bglighter;

$grey: #f8f8f2;
$blue: #6272a4;
$cyan: #8be9fd;
$green: #3CDD65;
$orange: #ffb86c;
$pink: #FC83D3;
$purple: #bd93f9;
$red: #ff5555;
$yellow: #f1fa8c;
$white: #fafafa;
$black: #202020;

$very-short-duration: 0.1s;
$short-duration: 0.3s;
$medium-duration: 0.5s;
$long-duration: 1s;

$mid_screen: 800px;
$small_screen: 600px;

@mixin roundCorners {
  border-radius: 0.25em;
}

@mixin circle {
  border-radius: 9999px;
  overflow: hidden;
}

@mixin transition {
  transition: 0.25s;
}

@mixin center {
  align-items: center;
  justify-content: center;
}

@mixin ellipsis {
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
}

@mixin shadow {
  box-shadow: 0 0 38px rgba(0, 0, 0, 0.3), 0 0 12px rgba(0, 0, 0, 0.22) !important;
}

.nuclear {
  font-family: "Lato", sans-serif;
  color: $white;
}

@keyframes fade-in {
  from {
    opacity: 0;
  }
  to {
    opacity: 1;
  }
}

@keyframes fade-out {
  from {
    opacity: 1;
  }
  to {
    opacity: 0;
  }
}
