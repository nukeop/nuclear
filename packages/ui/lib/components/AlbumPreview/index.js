import React from 'react';
import PropTypes from 'prop-types';
import _ from 'lodash';
import {
  withHandlers,
  withState,
  compose
} from 'recompose';
import { Button } from 'semantic-ui-react';

import { getThumbnail } from '../../utils';
import ContextPopup from '../ContextPopup';
import TrackRow from '../TrackRow';
import artPlaceholder from '../../../resources/media/art_placeholder.png';
import styles from './styles.scss';

const AlbumPreview = ({
  album,
  trackButtons,
  target,
  handleListClick,

  handleAddToQueue,
  handlePlayAll
}) => {
  const thumb = _.defaultTo(getThumbnail(album), artPlaceholder);
  const TrackButtons = trackButtons;

  return (
    <div className={styles.album_preview} >
      <div
        className={styles.album_cover}
        style={{ backgroundImage: `url(${thumb})` }}
      >
        <div className={styles.album_cover_overlay}>
          <Button
            basic
            icon='plus'
            size='huge'
            onClick={handleAddToQueue}
          />
          <Button
            basic
            icon='play'
            size='huge'
            onClick={handlePlayAll}
          />
        </div>
      </div>
      <div
        className={styles.track_list}
        onClick={handleListClick}
      >
        <table>
          <tbody>
            {
              album.tracks.map((track, index) => (
                <ContextPopup
                  trigger={
                    <TrackRow
                      key={index}
                      track={track}
                      mini
                      displayArtist
                      displayAlbum
                      displayTrackNumber
                      displayDuration
                    />
                  }
                  key={index}
                  thumb={thumb}
                  title={track.name}
                  artist={_.get(album, 'artist.name')}
                  target={target}
                >
                  <TrackButtons
                    track={track}
                    withAddToDownloads={false}
                  />
                </ContextPopup>
              ))
            }
          </tbody>
        </table>
      </div>
    </div>
  );
};

AlbumPreview.propTypes = {
  album: PropTypes.shape({
    artist: PropTypes.shape({
      name: PropTypes.string
    }),
    title: PropTypes.string.isRequired,
    tracks: PropTypes.arrayOf(PropTypes.shape({
      album: PropTypes.string,
      artist: PropTypes.shape({
        name: PropTypes.string
      }),
      name: PropTypes.string,
      duration: PropTypes.number,
      position: PropTypes.number,
      playcount: PropTypes.number
    })),
    image: PropTypes.arrayOf(PropTypes.shape({
      '#text': PropTypes.string
    }))
  }),
  trackButtons: PropTypes.object
};

export default compose(
  withState(
    'target', 'setTarget', { x: 0, y: 0 }
  ),
  withHandlers({
    handleListClick: ({ setTarget }) => e => setTarget({ x: e.clientX, y: e.clientY })
  })
)(AlbumPreview);
