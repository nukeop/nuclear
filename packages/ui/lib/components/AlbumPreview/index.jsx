import React from 'react';
import PropTypes from 'prop-types';
import _ from 'lodash';
import { Button } from 'semantic-ui-react';

import { getThumbnail } from '../..';
import ContextPopup from '../ContextPopup';
import TrackRow from '../TrackRow';
import artPlaceholder from '../../../resources/media/art_placeholder.png';
import styles from './styles.scss';

const AlbumPreview = ({
  album,
  trackButtons,

  handleAddToQueue,
  handlePlayAll
}) => {
  const thumb = _.defaultTo(getThumbnail(album), artPlaceholder);
  const TrackButtons = trackButtons;

  return (
    <div className={styles.album_preview} >
      <div
        className={styles.album_cover}
        style={{ backgroundImage: `url(${thumb})` }}
      >
        <div className={styles.album_cover_overlay}>
          <Button
            basic
            icon='plus'
            size='huge'
            onClick={handleAddToQueue}
          />
          <Button
            basic
            icon='play'
            size='huge'
            onClick={handlePlayAll}
          />
        </div>
      </div>
      <div
        className={styles.track_list}
      >
        <table>
          <tbody>
            {
              album.tracks.map((track, index) => (
                <ContextPopup
                  trigger={
                    <TrackRow
                      key={index}
                      track={track}
                      mini
                      displayArtist
                      displayAlbum
                      displayTrackNumber
                      displayDuration
                    />
                  }
                  key={index}
                  thumb={thumb}
                  title={track.name}
                  artist={_.get(album, 'artist.name')}
                >
                  <TrackButtons
                    track={track}
                    withAddToDownloads={false}
                  />
                </ContextPopup>
              ))
            }
          </tbody>
        </table>
      </div>
    </div>
  );
};

AlbumPreview.propTypes = {
  album: PropTypes.shape({
    artist: PropTypes.shape({
      name: PropTypes.string
    }),
    title: PropTypes.string.isRequired,
    tracks: PropTypes.arrayOf(PropTypes.shape({
      album: PropTypes.string,
      artist: PropTypes.shape({
        name: PropTypes.string
      }),
      name: PropTypes.string,
      duration: PropTypes.number,
      position: PropTypes.number,
      playcount: PropTypes.number
    })),
    image: PropTypes.arrayOf(PropTypes.shape({
      '#text': PropTypes.string
    }))
  }),
  trackButtons: PropTypes.object
};

export default AlbumPreview;
