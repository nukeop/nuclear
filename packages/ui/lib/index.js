import '../resources/Lato.css';

export { default as AlbumGrid } from './components/AlbumGrid';
export { default as AlbumPreview } from './components/AlbumPreview';
export { default as Card } from './components/Card';
export { default as ContextPopup } from './components/ContextPopup';
export { default as Cover } from './components/Cover';
export { default as SmoothImage } from './components/SmoothImage';
export { default as LibraryListTypeToggle } from './components/LibraryListTypeToggle';
export { default as Loader } from './components/Loader';
export { default as PopupButton } from './components/PopupButton';
export { default as QueueItem } from './components/QueueItem';
export { default as QueuePopup } from './components/QueuePopup';
export { default as Range } from './components/Range';
export { default as Seekbar } from './components/Seekbar';
export { default as ToastContainer } from './components/ToastContainer';
export { default as TrackRow } from './components/TrackRow';
export { default as UserPluginsItem } from './components/UserPluginsItem';

export * from './utils';
