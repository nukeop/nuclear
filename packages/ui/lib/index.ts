import '../resources/Lato.scss';

export { default as AlbumGrid } from './components/AlbumGrid';
export { default as ArtistGrid } from './components/ArtistGrid';
export { default as AlbumPreview } from './components/AlbumPreview';
export { default as Button } from './components/Button';
export { default as Card } from './components/Card';
export { default as Checkbox } from './components/Checkbox';
export { default as ContextPopup } from './components/ContextPopup';
export { default as Cover } from './components/Cover';
export { default as Dropdown } from './components/Dropdown';
export { default as LibraryListTypeToggle } from './components/LibraryListTypeToggle';
export { default as Loader } from './components/Loader';
export { default as MiniPlayer } from './components/MiniPlayer';
export { default as PlayerBar } from './components/PlayerBar';
export { default as PlayerButton } from './components/PlayerButton';
export { default as PopupButton } from './components/PopupButton';
export { default as PopupDropdown } from './components/PopupDropdown';
export { default as QueueItem } from './components/QueueItem';
export { default as Range } from './components/Range';
export { default as SearchBox } from './components/SearchBox';
export { default as Seekbar } from './components/Seekbar';
export { default as SmoothImage } from './components/SmoothImage';
export { default as StreamInfo } from './components/StreamInfo';
export { default as ToastContainer } from './components/ToastContainer';
export { default as Tooltip } from './components/Tooltip';
export { default as TrackPopup } from './components/TrackPopup';
export { default as TrackRow } from './components/TrackRow';
export { default as TrackTable } from './components/TrackTable';
export { default as UserPluginsItem } from './components/UserPluginsItem';
export { default as Visualizer } from './components/Visualizer';
export { default as VisualizerOverlay } from './components/VisualizerOverlay';

export * from './utils';
