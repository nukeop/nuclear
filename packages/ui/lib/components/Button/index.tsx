import React from 'react';
import cx from 'classnames';
import {
  Button as SUIButton,
  ButtonProps as SUIButtonProps
} from 'semantic-ui-react';

import common from '../../common.scss';
import styles from './styles.scss';

export type ButtonProps = SUIButtonProps & {
  borderless?: boolean;
}
const Button: React.FC<ButtonProps> = ({
  borderless = false,
  className,
  ...rest
}) => <SUIButton
  className={cx(
    common.nuclear,
    styles.button,
    {
      [styles.borderless]: borderless
    },
    className
  )}
  {...rest}
/>;

export default Button;
