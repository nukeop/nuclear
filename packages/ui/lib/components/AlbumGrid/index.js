import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import _ from 'lodash';
import {Dimmer, Loader } from 'semantic-ui-react';
import { compose, withState, withHandlers } from 'recompose';

import AlbumPreview from '../AlbumPreview';
import Card from '../Card';
import common from '../../common.scss';
import styles from './styles.scss';

export const getThumbnail = album => {
  return _.get(album, 'images[0].uri',
    _.get(album, 'image[0][\'#text\']',
      _.get(album, 'thumb')));
};

const AlbumGrid = ({
  albums,
  onAlbumClick,
  selectedAlbum,
  loading,
  trackButtons,
  withArtistNames,
  withAlbumPreview
}) => (
  <div className={cx(
    common.nuclear,
    styles.album_grid,
    {[styles.loading]: loading}
  )} >
  <div className={styles.album_cards}>
  {
    !loading &&
    !_.isEmpty(albums) &&
    albums.map((album, i) => (
      <Card
      key={i}
      header={album.title}
      content={withArtistNames && _.get(album, 'artist.name')}
      image={getThumbnail(album)}
      onClick={() => onAlbumClick(album)}
      />
    ))
  }
  </div>

  {
    !loading && withAlbumPreview &&
      <>
        <hr />
        <AlbumPreview
          album={selectedAlbum}
          trackButtons={trackButtons}
        />
      </>
    }
    { loading && <Dimmer active><Loader /></Dimmer> }
  </div>
);

AlbumGrid.propTypes = {
  albums: PropTypes.array,
  onAlbumClick: PropTypes.func,
  loading: PropTypes.bool,
  withArtistNames: PropTypes.bool,
  withAlbumPreview: PropTypes.bool,

  trackButtons: PropTypes.arrayOf(PropTypes.node)
};

export default compose(
  withState(
    'selectedAlbum',
    'selectAlbum',
    ({ albums }) => _.head(albums)),
  withHandlers({
    onAlbumClick: ({ selectAlbum }) => album => selectAlbum(album)
  })
)(AlbumGrid);
