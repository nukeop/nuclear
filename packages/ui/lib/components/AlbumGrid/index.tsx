import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import _ from 'lodash';
import { Dimmer, Loader } from 'semantic-ui-react';
import { compose, withState, withHandlers } from 'recompose';

import AlbumPreview from '../AlbumPreview';
import Card from '../Card';
import common from '../../common.scss';
import styles from './styles.scss';
import { getThumbnail, getTrackItem } from '../..';

const AlbumGrid = ({
  albums,
  onAlbumClick,
  removeFavoriteAlbum,
  selectedAlbum,
  loading,
  trackButtons,
  onAddToQueue,
  onPlayAll,
  autoSize,
  withArtistNames,
  withAlbumPreview
}) => (
  <div className={cx(
    common.nuclear,
    styles.album_grid,
    { [styles.loading]: loading },
    { [styles.auto_size]: autoSize && !loading }
  )} >
    <div className={styles.album_cards}>
      {
        !loading &&
          !_.isEmpty(albums) &&
          albums.map((album, i) => (
            <Card
              key={i}
              header={album.title}
              content={withArtistNames && _.get(album, 'artist.name')}
              image={getThumbnail(album)}
              onClick={() => onAlbumClick(album)}
              withMenu={removeFavoriteAlbum ? true : false}
              menuEntries={[
                {
                  type: 'item', props: {
                    children: 'Remove',
                    onClick: () => removeFavoriteAlbum(album)
                  }
                }
              ]}
            />
          ))
      }
    </div>

    {
      !loading && withAlbumPreview &&
        <Fragment>
          <hr />
          <AlbumPreview
            album={selectedAlbum}
            trackButtons={trackButtons}
            handleAddToQueue={onAddToQueue}
            handlePlayAll={onPlayAll}
          />
        </Fragment>
    }
    {loading && <Dimmer active><Loader /></Dimmer>}
  </div>
);

AlbumGrid.propTypes = {
  albums: PropTypes.array,
  streamProviders: PropTypes.array, // eslint-disable-line react/no-unused-prop-types
  onAlbumClick: PropTypes.func,
  removeFavoriteAlbum: PropTypes.func,
  loading: PropTypes.bool,
  addToQueue: PropTypes.func, // eslint-disable-line react/no-unused-prop-types
  clearQueue: PropTypes.func, // eslint-disable-line react/no-unused-prop-types
  selectSong: PropTypes.func, // eslint-disable-line react/no-unused-prop-types
  startPlayback: PropTypes.func, // eslint-disable-line react/no-unused-prop-types
  onAddToQueue: PropTypes.func,
  onPlayAll: PropTypes.func,

  autoSize: PropTypes.bool,
  withArtistNames: PropTypes.bool,
  withAlbumPreview: PropTypes.bool,

  trackButtons: PropTypes.object
};

export default compose(
  withState(
    'selectedAlbum',
    'selectAlbum',
    ({ albums }) => _.head(albums)
  ),
  withHandlers({
    onAlbumClick: ({ onAlbumClick, selectAlbum }) => album => _.isNil(onAlbumClick) ? selectAlbum(album) : onAlbumClick(album),
    onAddToQueue: ({ addToQueue, selectedAlbum }) => () => {
      selectedAlbum.tracks.map(track => addToQueue(getTrackItem(track)));
    }
  }),
  withHandlers({
    onPlayAll: ({ selectSong, startPlayback, onAddToQueue }) => () => {
      onAddToQueue();
      selectSong(0);
      startPlayback();
    }
  })
)(AlbumGrid);
