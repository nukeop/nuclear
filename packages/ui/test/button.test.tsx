import { Button } from '../lib';
import { ButtonProps } from '../lib/components/Button';
import { makeSnapshotTest } from './helpers';

const propsList: ButtonProps[] = [
  { basic: true },
  {},
  { color: 'green' },
  { color: 'blue' },
  { color: 'purple' },
  { color: 'pink' },
  { color: 'orange' },
  { color: 'red' }
];

propsList.forEach(props => {
  makeSnapshotTest(Button, { ...props, children: 'Test button' }, `(Snapshot) Button - ${JSON.stringify(props)}`);
  makeSnapshotTest(Button, { ...props, circular: true, children: 'Test button' }, `(Snapshot) Button - circular - ${JSON.stringify(props)}`);
  makeSnapshotTest(Button, { ...props, icon: 'play' }, `(Snapshot) Button - icon - ${JSON.stringify(props)}`);
  makeSnapshotTest(Button, { ...props, circular: true, icon: 'play' }, `(Snapshot) Button - circular icon - ${JSON.stringify(props)}`);
});
