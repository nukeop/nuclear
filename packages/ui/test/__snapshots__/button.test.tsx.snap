// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Snapshot) Button - {"basic":true} should render correctly 1`] = `
<DocumentFragment>
  <button
    class="ui basic button nuclear button"
  >
    Test button
  </button>
</DocumentFragment>
`;

exports[`(Snapshot) Button - {"color":"blue"} should render correctly 1`] = `
<DocumentFragment>
  <button
    class="ui blue button nuclear button"
  >
    Test button
  </button>
</DocumentFragment>
`;

exports[`(Snapshot) Button - {"color":"green"} should render correctly 1`] = `
<DocumentFragment>
  <button
    class="ui green button nuclear button"
  >
    Test button
  </button>
</DocumentFragment>
`;

exports[`(Snapshot) Button - {"color":"orange"} should render correctly 1`] = `
<DocumentFragment>
  <button
    class="ui orange button nuclear button"
  >
    Test button
  </button>
</DocumentFragment>
`;

exports[`(Snapshot) Button - {"color":"pink"} should render correctly 1`] = `
<DocumentFragment>
  <button
    class="ui pink button nuclear button"
  >
    Test button
  </button>
</DocumentFragment>
`;

exports[`(Snapshot) Button - {"color":"purple"} should render correctly 1`] = `
<DocumentFragment>
  <button
    class="ui purple button nuclear button"
  >
    Test button
  </button>
</DocumentFragment>
`;

exports[`(Snapshot) Button - {"color":"red"} should render correctly 1`] = `
<DocumentFragment>
  <button
    class="ui red button nuclear button"
  >
    Test button
  </button>
</DocumentFragment>
`;

exports[`(Snapshot) Button - {} should render correctly 1`] = `
<DocumentFragment>
  <button
    class="ui button nuclear button"
  >
    Test button
  </button>
</DocumentFragment>
`;

exports[`(Snapshot) Button - circular - {"basic":true} should render correctly 1`] = `
<DocumentFragment>
  <button
    class="ui basic circular button nuclear button"
  >
    Test button
  </button>
</DocumentFragment>
`;

exports[`(Snapshot) Button - circular - {"color":"blue"} should render correctly 1`] = `
<DocumentFragment>
  <button
    class="ui blue circular button nuclear button"
  >
    Test button
  </button>
</DocumentFragment>
`;

exports[`(Snapshot) Button - circular - {"color":"green"} should render correctly 1`] = `
<DocumentFragment>
  <button
    class="ui green circular button nuclear button"
  >
    Test button
  </button>
</DocumentFragment>
`;

exports[`(Snapshot) Button - circular - {"color":"orange"} should render correctly 1`] = `
<DocumentFragment>
  <button
    class="ui orange circular button nuclear button"
  >
    Test button
  </button>
</DocumentFragment>
`;

exports[`(Snapshot) Button - circular - {"color":"pink"} should render correctly 1`] = `
<DocumentFragment>
  <button
    class="ui pink circular button nuclear button"
  >
    Test button
  </button>
</DocumentFragment>
`;

exports[`(Snapshot) Button - circular - {"color":"purple"} should render correctly 1`] = `
<DocumentFragment>
  <button
    class="ui purple circular button nuclear button"
  >
    Test button
  </button>
</DocumentFragment>
`;

exports[`(Snapshot) Button - circular - {"color":"red"} should render correctly 1`] = `
<DocumentFragment>
  <button
    class="ui red circular button nuclear button"
  >
    Test button
  </button>
</DocumentFragment>
`;

exports[`(Snapshot) Button - circular - {} should render correctly 1`] = `
<DocumentFragment>
  <button
    class="ui circular button nuclear button"
  >
    Test button
  </button>
</DocumentFragment>
`;

exports[`(Snapshot) Button - circular icon - {"basic":true} should render correctly 1`] = `
<DocumentFragment>
  <button
    class="ui basic circular icon button nuclear button"
  >
    <i
      aria-hidden="true"
      class="play icon"
    />
  </button>
</DocumentFragment>
`;

exports[`(Snapshot) Button - circular icon - {"color":"blue"} should render correctly 1`] = `
<DocumentFragment>
  <button
    class="ui blue circular icon button nuclear button"
  >
    <i
      aria-hidden="true"
      class="play icon"
    />
  </button>
</DocumentFragment>
`;

exports[`(Snapshot) Button - circular icon - {"color":"green"} should render correctly 1`] = `
<DocumentFragment>
  <button
    class="ui green circular icon button nuclear button"
  >
    <i
      aria-hidden="true"
      class="play icon"
    />
  </button>
</DocumentFragment>
`;

exports[`(Snapshot) Button - circular icon - {"color":"orange"} should render correctly 1`] = `
<DocumentFragment>
  <button
    class="ui orange circular icon button nuclear button"
  >
    <i
      aria-hidden="true"
      class="play icon"
    />
  </button>
</DocumentFragment>
`;

exports[`(Snapshot) Button - circular icon - {"color":"pink"} should render correctly 1`] = `
<DocumentFragment>
  <button
    class="ui pink circular icon button nuclear button"
  >
    <i
      aria-hidden="true"
      class="play icon"
    />
  </button>
</DocumentFragment>
`;

exports[`(Snapshot) Button - circular icon - {"color":"purple"} should render correctly 1`] = `
<DocumentFragment>
  <button
    class="ui purple circular icon button nuclear button"
  >
    <i
      aria-hidden="true"
      class="play icon"
    />
  </button>
</DocumentFragment>
`;

exports[`(Snapshot) Button - circular icon - {"color":"red"} should render correctly 1`] = `
<DocumentFragment>
  <button
    class="ui red circular icon button nuclear button"
  >
    <i
      aria-hidden="true"
      class="play icon"
    />
  </button>
</DocumentFragment>
`;

exports[`(Snapshot) Button - circular icon - {} should render correctly 1`] = `
<DocumentFragment>
  <button
    class="ui circular icon button nuclear button"
  >
    <i
      aria-hidden="true"
      class="play icon"
    />
  </button>
</DocumentFragment>
`;

exports[`(Snapshot) Button - icon - {"basic":true} should render correctly 1`] = `
<DocumentFragment>
  <button
    class="ui basic icon button nuclear button"
  >
    <i
      aria-hidden="true"
      class="play icon"
    />
  </button>
</DocumentFragment>
`;

exports[`(Snapshot) Button - icon - {"color":"blue"} should render correctly 1`] = `
<DocumentFragment>
  <button
    class="ui blue icon button nuclear button"
  >
    <i
      aria-hidden="true"
      class="play icon"
    />
  </button>
</DocumentFragment>
`;

exports[`(Snapshot) Button - icon - {"color":"green"} should render correctly 1`] = `
<DocumentFragment>
  <button
    class="ui green icon button nuclear button"
  >
    <i
      aria-hidden="true"
      class="play icon"
    />
  </button>
</DocumentFragment>
`;

exports[`(Snapshot) Button - icon - {"color":"orange"} should render correctly 1`] = `
<DocumentFragment>
  <button
    class="ui orange icon button nuclear button"
  >
    <i
      aria-hidden="true"
      class="play icon"
    />
  </button>
</DocumentFragment>
`;

exports[`(Snapshot) Button - icon - {"color":"pink"} should render correctly 1`] = `
<DocumentFragment>
  <button
    class="ui pink icon button nuclear button"
  >
    <i
      aria-hidden="true"
      class="play icon"
    />
  </button>
</DocumentFragment>
`;

exports[`(Snapshot) Button - icon - {"color":"purple"} should render correctly 1`] = `
<DocumentFragment>
  <button
    class="ui purple icon button nuclear button"
  >
    <i
      aria-hidden="true"
      class="play icon"
    />
  </button>
</DocumentFragment>
`;

exports[`(Snapshot) Button - icon - {"color":"red"} should render correctly 1`] = `
<DocumentFragment>
  <button
    class="ui red icon button nuclear button"
  >
    <i
      aria-hidden="true"
      class="play icon"
    />
  </button>
</DocumentFragment>
`;

exports[`(Snapshot) Button - icon - {} should render correctly 1`] = `
<DocumentFragment>
  <button
    class="ui icon button nuclear button"
  >
    <i
      aria-hidden="true"
      class="play icon"
    />
  </button>
</DocumentFragment>
`;
